File: tests/rustdoc-js/reexport-dedup.js
// exact-check

const EXPECTED = [
    {
        'query': 'Subscriber',
        'others': [
            { 'path': 'foo', 'name': 'Subscriber' },
        ],
    },
    {
        'query': 'fmt Subscriber',
        'others': [
            { 'path': 'foo::fmt', 'name': 'Subscriber' },
        ],
    },
    {
        'query': 'AnotherOne',
        'others': [
            { 'path': 'foo', 'name': 'AnotherOne' },
        ],
    },
];

File: tests/rustdoc-js/reexport-dedup.rs
// This test enforces that the (renamed) reexports are present in the search results.
#![crate_name = "foo"]

pub mod fmt {
    pub struct Subscriber;
}
mod foo {
    pub struct AnotherOne;
}

pub use fmt::Subscriber;
pub use foo::AnotherOne;
